hugr-model/src/v0/mod.rs

```diff
@@ -664,6 +652,24 @@ pub enum Term<'a> {
     ControlType,
 }
 
+/// An item in a list term.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum ListItem {
+    /// A single item.
+    Item(TermId),
+    /// A list to be spliced into the parent list.
+    Splice(TermId),
```

Comment:
I guess the model-validity-checker that tests that this TermId refers to a thing of appropriate List type, whereas the TermId in a `ListItem::Item` refers to something of appropriate element type, does not exist yet??

Reply:
That's correct; this would be done by typechecking.